### ISSUE TITLE
[Bug]: Storybook 7.2.0 : Broken Build on clean new project 

### ISSUE THREAD

**Issue by Tisamu** [storybookjs/storybook#23694]
### Describe the bug

When trying to start Storybook 7.2.0 on a fresh vite+react project, Storybook won't start, juste showing a broken build message :

![](https://user-images.githubusercontent.com/15633268/257755681-0881274c-5260-4915-8f90-b94feffcada3.png)

### To Reproduce

Create new react project with vite, install & init Storybook 7.2.0 and try to start it.

### System

```shell
I can't from this computer, sorry. :/
```

### Additional context

I'm opening a new issue but it's the same as related in #8016.

As suggested in the #8016, I just created a new issue to make the problem visible but you may close this one and re-open the #8016.

**Comment by mamadshni** [comment]
the issue is with missing module `esbuild`

**Comment by martaver** [comment]
I've noticed a related problem with `7.2.0`, that build errors are no longer written to Terminal, making it impossible to debug such issues. `storybook` writes:

```
WARN Broken build, fix the error above.
WARN You may need to refresh the browser.
```

But there's no error described 'above'.

**Comment by miljan-aleksic** [comment]
The swallowed errors have been fixed, we must wait for next release, though.

https://github.com/storybookjs/storybook/pull/23689

**Comment by tax-k** [comment]
I get the same issue as @Tisamu

**Comment by vanessayuenn** [comment]
The fix for the errors getting swallowed has been released in as of 7.2.1. Can someone try it with the latest storybook version (7.2.1+) and see if they can report back with more error information? Thanks!
